### PR TITLE
Add jobset to k8s.gcr.io

### PR DESCRIPTION
**Pull request by danielvegamyhre**
We would like to host images for [jobset](https://github.com/kubernetes-sigs/jobset) in k8s.gcr.io, see https://github.com/kubernetes-sigs/jobset/issues/22

/sig apps

cc @ahg-g

### CODE REVIEW

File: k8s.gcr.io/images/k8s-staging-jobset/OWNERS
Comment (ahg-g) [inline k8s.gcr.io/images/k8s-staging-jobset/OWNERS:1]: it seems that this now should be under "k8s.io/registry.k8s.io/images"

File: k8s.gcr.io/images/k8s-staging-jobset/OWNERS
Comment (ahg-g) [inline k8s.gcr.io/images/k8s-staging-jobset/OWNERS:8]: add new line

File: k8s.gcr.io/images/k8s-staging-jobset/OWNERS
Comment (danielvegamyhre) [reply k8s.gcr.io/images/k8s-staging-jobset/OWNERS]: Deleted these and recreated them in `k8s.io/registry.k8s.io/images/k8s-staging/jobset`

**Comment by danielvegamyhre** [comment]
verify test error message is:

```
--- FAIL: TestManifestLooksReasonable (0.02s)
    manifest_test.go:43: did not find all production registries, 3 found but expected 23 for image "jobset" in "k8s-staging-jobset/promoter-manifest.yaml"
```

Not sure what to make of it yet

**Comment by ahg-g** [comment]
may be we need a cloudbuild file in the jobset repo, see example: https://github.com/kubernetes-sigs/kueue/pull/84 ?

**Comment by ahg-g** [comment]
@ArangoGutierrez any idea how to address this error?

**Comment by ahg-g** [comment]
/assign @ameukam 

ameukam any chance you can help with this one?

File: k8s.gcr.io/manifests/k8s-staging-jobset/promoter-manifest.yaml
Comment (ArangoGutierrez) [inline k8s.gcr.io/manifests/k8s-staging-jobset/promoter-manifest.yaml:2]: See https://github.com/kubernetes/k8s.io/blob/main/k8s.gcr.io/manifests/k8s-staging-kueue/promoter-manifest.yaml

you may be missing a couple

Comment (ArangoGutierrez) [reply k8s.gcr.io/manifests/k8s-staging-jobset/promoter-manifest.yaml:2]: you have 3 it expects 23 as seen on kueue example

Comment (ameukam) [reply k8s.gcr.io/manifests/k8s-staging-jobset/promoter-manifest.yaml:2]: Can confirm what @ArangoGutierrez said. more Artifact Registry repos are expected for promotion.

**Review by soltysh — APPROVED** [review]
/lgtm
/approve
from sig-apps pov

File: k8s.gcr.io/manifests/k8s-staging-jobset/promoter-manifest.yaml
Comment (ahg-g) [reply k8s.gcr.io/manifests/k8s-staging-jobset/promoter-manifest.yaml:2]: Thanks all! @ameukam this is ready

**Comment by ahg-g** [comment]
@ameukam friendly ping

**Comment by danielvegamyhre** [comment]
Seems like the label `do-not-merge/blocked-paths` is blocking the merge, which indicates that a PR should not merge because it touches files in blocked paths. I'm not sure what is required to get it removed, anyone have any insight on this?

File: groups/restrictions.yaml
Comment (ArangoGutierrez) [inline groups/restrictions.yaml:30]: This might be the file preventing the merge

**Review by ArangoGutierrez — CHANGES_REQUESTED** [review]

File: groups/restrictions.yaml
Comment (ahg-g) [reply groups/restrictions.yaml:30]: any idea how to address it? are we not supposed to this?

File: k8s.gcr.io/manifests/k8s-staging-jobset/promoter-manifest.yaml
Comment (ahg-g) [inline k8s.gcr.io/manifests/k8s-staging-jobset/promoter-manifest.yaml:1]: @danielvegamyhre this file should move to `registry.k8s.io` instead of `k8s.gcr.io`

File: groups/restrictions.yaml
Comment (ahg-g) [reply groups/restrictions.yaml:30]: actually, Bob pointed out that the culprit is [k8s.gcr.io/manifests/k8s-staging-jobset/promoter-manifest.yaml](https://github.com/kubernetes/k8s.io/pull/5144/files#diff-5ea082290b72efae9bf40dd30f2d1e4ed8a079fa440cbacb57a332fbb360aed0) which should move to registry.k8s.io/...

File: k8s.gcr.io/manifests/k8s-staging-jobset/promoter-manifest.yaml
Comment (ahg-g) [reply k8s.gcr.io/manifests/k8s-staging-jobset/promoter-manifest.yaml:1]: so it should be registry.k8s.io/manifests/k8s-staging-jobset/promoter-manifest.yaml

Comment (danielvegamyhre) [reply k8s.gcr.io/manifests/k8s-staging-jobset/promoter-manifest.yaml]: Ah ok I see. Done.

**Comment by danielvegamyhre** [comment]
@nikhita @ArangoGutierrez @soltysh LGTMs need to be re-applied when you have a moment please, since I had to change files to resolve the issue `do-not-merge/blocked-paths` label issue (by moving the promoter manifest under `registry.k8s.io/manifests` instead of `k8s.gcr.io/manifests`).

**Comment by BenTheElder** [comment]
https://github.com/kubernetes/k8s.io/tree/main/registry.k8s.io#creating-staging-repos

**Comment by danielvegamyhre** [comment]
@BenTheElder the title of the PR is outdated, the changes themselves are under registry.8s.io. I don't think we've created the staging repo yet, I'll take a look at that, thanks!

**Comment by danielvegamyhre** [comment]
@BenTheElder I created the staging repo [here](https://github.com/kubernetes/k8s.io/pull/5144/files#diff-e07327d279a7e6af82d2be88e98c5d3e02c8b0682257d0f69ff7122905443384R302) but I don't think this step has been performed yet: 

"After the PR is merged ... whoever approved your PR will run [the necessary bash script(s)](https://github.com/kubernetes/k8s.io/blob/main/infra/gcp/bash/ensure-staging-storage.sh) to create the staging repo"

**Comment by BenTheElder** [comment]
Ah, I think @nikhita was able to approve this but may not actually have the infra access, which I think is just the SIG K8s leads.

@ameukam, @dims, @thockin can do this.

**Comment by dims** [comment]
i am on it!

**Comment by dims** [comment]
```
-release-viewers@kubernetes.io as project viewers in k8s-staging-ci-images 
  Empowering k8s-infra-release-viewers@kubernetes.io as project viewers in k8s-staging-cip-test 
  Empowering k8s-infra-release-viewers@kubernetes.io as project viewers in k8s-staging-experimental 
  Empowering k8s-infra-release-viewers@kubernetes.io as project viewers in k8s-staging-kubernetes 
  Empowering kubernetes-release-test GCB service account to admin GCR 
  Empowering k8s-infra-release-viewers@kubernetes.io as project viewers in k8s-staging-releng 
  Empowering k8s-infra-release-viewers@kubernetes.io as project viewers in k8s-staging-releng-test 
  Empowering k8s-infra-release-viewers@kubernetes.io as project viewers in k8s-staging-publishing-bot 
Done 
```

**Pull request merged**
